library/ssl_tls13_client.c

```diff
@@ -4104,28 +4118,13 @@ int mbedtls_ssl_handshake_client_step( mbedtls_ssl_context *ssl )
                 mbedtls_ack_add_record( ssl, MBEDTLS_SSL_HS_FINISHED, MBEDTLS_SSL_ACK_RECORDS_RECEIVED );
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
 
-#if defined(MBEDTLS_ZERO_RTT)
-            if( ssl->handshake->early_data == MBEDTLS_SSL_EARLY_DATA_ON )
-            {
-                mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_EARLY_DATA );
-            }
-            else
-#endif /* MBEDTLS_ZERO_RTT */
-            {
-#if defined(MBEDTLS_SSL_TLS13_COMPATIBILITY_MODE)
-                mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_CLIENT_CCS_AFTER_SERVER_FINISHED );
-#else
-                mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_CLIENT_CERTIFICATE );
-#endif /* MBEDTLS_SSL_TLS13_COMPATIBILITY_MODE */
-
-            }
-            break;
```

Comment:
@hanno-arm , Is this intentional of not using break here? The case of `MBEDTLS_SSL_SERVER_FINISHED` will fall through to other underlying cases.

Reply:
No, that wasn't intentional. It accidentally happens to work, but it's not right this way. Thanks for spotting!